sktime/transformations/series/fourier.py

```diff
@@ -174,6 +203,10 @@ def _transform(self, X, y=None):
         transformed version of X
         """
         X_transformed = deepcopy(X)
+
+        if isinstance(X.index, pd.DatetimeIndex):
+            X.index = X.index.to_period(self.freq)
```

Comment:
use the new attribute `self.freq_` described above instead and do the operation on X_transformed that at this point is a deepcopy of X to avoid any side effects on X.